src/tools/monaco/validate.ts

```diff
@@ -26,15 +26,17 @@ export function validateDSL(monaco: typeof MonacoEditor, dsl: string): Marker[]
       }
 
       const extraInformation: Marker["extraInformation"] = {};
-      const errorMetadata = singleErr.metadata as any;
-      if (errorMetadata.errorType) {
-        extraInformation.error = errorMetadata.errorType;
-      }
-      ["typeName", "relation"].forEach((field) => {
-        if (errorMetadata[field]) {
-          (extraInformation as any)[field] = errorMetadata[field];
+      const errorMetadata = singleErr.metadata;
+      if (errorMetadata) {
+        if ("errorType" in errorMetadata) {
```

Comment:
This is to work around TS reporting that `errorType` doesn't exist on all types for `errorMetadata`, given that we have checked that `errorMetadata` exists this is safe.

It would be nice to use a type predicate here to improve readability but we don't have the types for `metadata` exposed from language so can't really do that.

Reply:
> we don't have the types for metadata exposed from language

Can you add a ticket there so that we can remember to do that?

Reply:
Filed https://github.com/openfga/language/issues/140 on the language repo